pkg/csi/service/wcp/controller.go

```diff
@@ -538,135 +538,166 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
+		vmuuid, err := getVMUUIDFromK8sCloudOperatorService(ctx, req.VolumeId, req.NodeId)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get the pod vmuuid annotation from the k8sCloudOperator service when processing attach for volumeID: %s on node: %s. Error: %+v", req.VolumeId, req.NodeId, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
 
-	// Connect to VC
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	err = vc.Connect(ctx)
-	if err != nil {
-		msg := fmt.Sprintf("failed to connect to Virtual Center: %s", vc.Config.Host)
-		log.Error(msg)
-		return nil, status.Errorf(codes.Internal, msg)
-	}
+		vcdcMap, err := getDatacenterFromConfig(c.manager.CnsConfig)
+		if err != nil {
+			msg := fmt.Sprintf("failed to get datacenter from config with error: %+v", err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		var vCenterHost, dcMorefValue string
+		for key, value := range vcdcMap {
+			vCenterHost = key
+			dcMorefValue = value
+		}
+		vc, err := c.manager.VcenterManager.GetVirtualCenter(ctx, vCenterHost)
+		if err != nil {
+			msg := fmt.Sprintf("Cannot get virtual center %s from virtualcentermanager while attaching disk with error %+v",
+				vc.Config.Host, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
 
-	podVM, err := getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, vmuuid)
-	if err != nil {
-		msg := fmt.Sprintf("failed to the PodVM Moref from the PodVM UUID: %s in datacenter: %s with err: %+v", vmuuid, dcMorefValue, err)
-		log.Error(msg)
-		return nil, status.Errorf(codes.Internal, msg)
-	}
+		// Connect to VC
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		err = vc.Connect(ctx)
+		if err != nil {
+			msg := fmt.Sprintf("failed to connect to Virtual Center: %s", vc.Config.Host)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
 
-	// Attach the volume to the node
-	diskUUID, err := common.AttachVolumeUtil(ctx, c.manager, podVM, req.VolumeId)
-	if err != nil {
-		if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FakeAttach) {
-			log.Infof("Volume attachment failed. Checking if it can be fake attached")
-			var capabilities []*csi.VolumeCapability
-			capabilities = append(capabilities, req.VolumeCapability)
-			if !common.IsFileVolumeRequest(ctx, capabilities) { //Block volume
-				allowed, err := commonco.ContainerOrchestratorUtility.IsFakeAttachAllowed(ctx, req.VolumeId, c.manager.VolumeManager)
-				if err != nil {
-					msg := fmt.Sprintf("failed to determine if volume: %s can be fake attached. Error: %+v", req.VolumeId, err)
-					log.Error(msg)
-					return nil, status.Errorf(codes.Internal, msg)
-				}
+		podVM, err := getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, vmuuid)
+		if err != nil {
+			msg := fmt.Sprintf("failed to the PodVM Moref from the PodVM UUID: %s in datacenter: %s with err: %+v", vmuuid, dcMorefValue, err)
```

Comment:
not related to this PR, `failed to the PodVM Moref` ==> `failed to fetch the PodVM Moref`

Reply:
The PR is already merged. I can take care of it in my next PR.